data_cleaner.do

```diff
@@ -0,0 +1,274 @@
+	by market: strgroup clean_hcity, gen(grouped_hcity) threshold(0.25) first 
+
+	// Generate frequency of the groups
+	bysort grouped_hcity: egen group_freq = count(grouped_hcity)
+	keep clean_hcity grouped_hcity group_freq
+	// Save the data to a temporary file
+	tempfile temp
+	save "`temp'", replace 
+	restore
+	// Merge by clean_hcity using the temporary file
+	merge m:1 clean_hcity using "`temp'"
+	drop _merge
+		
+	gen grouping_hcity = .
+	// Replace grouping_hcity with grouped_hcity if group_freq is greater than or equal to 2
+	replace grouping_hcity = grouped_hcity if group_freq>=2
+	// Find the mode of clean_hcity by grouping_hcity
+	bysort grouping_hcity: egen group_mode = mode(clean_hcity) 
+	// Replace final_city with group_mode if group_freq is greater than or equal to 2 and final_city is "."
+	replace final_city = group_mode if (group_freq >=2) & (final_city == ".")
+	// Set good_city to 1 if group_freq is greater than or equal to 2
+	replace good_city=1 if group_freq >=2
+		
+	// Loop matchit through other accepts
+	forvalues i = 1 /4{
+		matchit clean_hcity clean_allaccept`i', similmethod(soundex_ext) weights(log)
+		// Replace final_city with clean_primary if similscore is greater than 0 and final_city is "."
+		replace final_city = clean_primary if (similscore > 0) & (final_city == ".")
+		// Set good_city to 1 if similscore is greater than 0
+		replace good_city = 1 if similscore > 0
+		// Drop similscore
+		drop similscore
+	}
+
+
+// Section 4: Manual changes
+    // Create a list of pairs of strings to replace with the format "ORIGINAL_VALUE>NEW_VALUE"
+	local manual_changes "CAROLLTON>CARROLLTON " ///
+	"SMRYNA>SMYRNA " ///
+	"ANN>ANNAPOLIS " ///
+	"COLUMBIAMD>COLUMBIA " ///
+	"NBELAIR>BELAIR " ///
+	"MIDDLEBURGHTS>MIDDLEBURGHEIGHTS " ///
+	"MORTHRIDGEVILLE>NORTHRIDGEVILLE " ///
+	"WOODBRIGE>WOODBRIDGE " ///
+	"LEWSIVILLE>LEWISVILLE " ///
+	"SACSHE>SACHSE " ///
+	"MACOMBTWP>MACOMB " ///
+	"MACOMBTOWNSHIP>MACOMB " ///
+	"SANANTONIOTX>SANANTONIO " ///
+	"NORTHRICHLANDS>NORTHRICHLANDHILLS " ///
+	"NRICHLANDHILLS>NORTHRICHLANDHILLS " ///
+	"HOUS>HOUSTON " ///
+	"HUOSTON>HOUSTON " ///
+	"BLETON>BELTON " ///
+	"OVERLAND>OVERLANDPARK " ///
+	"NORTHLONGBEACH>LONGBEACH " ///
+	"EASTAMWELL>EASTAMWELLTWP " ///
+	"GLENBROOKE>GREENBROOK " ///
+	"MONJUNCTION>MONMOUTHJUNCTION " ///
+	"NORTHEDISON>EDISON " ///
+	"SOUTHEDISON>EDISON " ///
+	"WOODBRIDGEPROPER>WOODBRIDGE " ///
+	"BLOOMFIELDTWP>BLOOMFIELD " ///
+	"BOONTONTOWN>BOONTON " ///
+	"CALDWELLBOROTWP>CALDWELL " ///
+	"CHATHAMBOROUGH>CHATHAM " ///
+	"CHATHAMTOWNSHIP>CHATHAM " ///
+	"HANOVERTWP>HANOVERTOWNSHIP " ///
+	"MADISONBOROUGH>MADISON " ///
+	"MORRISTOWNSHIP>MORRISTOWN " ///
+	"MORRISTWP>MORRISTOWN " ///
+	"MOUNTOLIVETWP>MOUNTOLIVE " ///
+	"MTOLIVE>MOUNTOLIVE " ///
+	"MTOLIVETWP>MOUNTOLIVE " ///
+	"PARSIPPANYTROYHILLS>PARSIPPANY " ///
+	"BAYRIDGEBROOKLYN>BROOKLYN " ///
+	"BKLYN>BROOKLYN " ///
+	"FLUSHINGQUEENS>FLUSHING " ///
```

Comment:
I think that `BAYRIDGEBROOKLYN` , `BKLYN`, and `FLUSHINGQUEENS` should all be set to whatever New York City is. Brooklyn and Queens are borroughs, NYC would be more consistent.

Reply:
This is a great point, and in making the change I realized that we actually have a lot of the boroughs as final city names in the current version of our cleaning code. Because each borough is so big, and the housing markets I assume are different, and we had good data on each of them, I though it could be reasonable to use the boroughs as the final unit of analysis. But the issue was what to do with entries that had just been marked as "New York" for city. So I did some digging and manual inspection and found that all entries that I checked (about 20) that were ending up as NEWYORK in the final version of the cleaning code were actually houses in Manhattan (which is colloquially sometimes just called New York City). So I decided to mark all New York houses with their boroughs for consistency, changing NEWYORK to MANHATTAN in all instances.